src/lib/message_filters.py

```diff
@@ -75,12 +76,33 @@ def filter_time_range(messages, time_key, start_time_inclusive, end_time_inclusi
         :return: Filtered list.
         :rtype: list of TracedData
         """
+        # De-duplicate time_keys
+        time_keys = set(time_keys)
```

Comment:
consider asserting that it's a set instead, given the type declaration above?

Reply:
changed